packages/signers/multisig-signer/src/multisig-signer.ts

```diff
@@ -0,0 +1,74 @@
+import { getDynamicBuilder, getLookupFn } from "@polkadot-api/metadata-builders"
+import type { PolkadotSigner } from "@polkadot-api/polkadot-signer"
+import {
+  decAnyMetadata,
+  getSs58AddressInfo,
+  SS58String,
+} from "@polkadot-api/substrate-bindings"
+import { toHex } from "@polkadot-api/utils"
+
+export function multisigSigner(
+  multisig: {
+    threshold: number
+    signatories: SS58String[]
+  },
+  signer: PolkadotSigner,
+): PolkadotSigner {
+  const publicAddr = toHex(signer.publicKey)
+  const otherSignatories = multisig.signatories.filter((addr) => {
+    const info = getSs58AddressInfo(addr)
+    if (!info.isValid) throw new Error("Invalid address " + addr)
+    return toHex(info.publicKey) !== publicAddr
+  })
+  if (otherSignatories.length === multisig.signatories.length) {
+    throw new Error("Signer is not one of the signatories of the multisig")
+  }
```

Comment:
I'm not sure if this is the best option or if the `other_signatories` option is better. That's how it is in the extrinsic. I'm on the fence.
Maybe renaming the `signatories` parameter to `allSignatories` makes it clearer for the consumer of the library.